File: inst/Julia/pO2.jl
function pO2(TC,atm=760)
    
    if TC < 0 || TC > 40
        return println("Temp beyond valid model range 0-40 C") 
    end
    
    function HC(TC)
    return(-0.0000058333*TC^3+0.0001821*TC^2+0.072405*TC+2.5443)*10000
    end
    
    function vp(TC) 
    0.0456*TC^2-0.8559*TC+16.509
    end
    
    function coef(TC)
       if  TC<30 
            coef=.678
        else
            coef = .827
        end
        return coef
    end
    
    function adj(TC)
        
      if  TC<30 
            adj = 35
        else
            adj = 49
        end
        return adj  
        
    end
    
    function DO(TC,vp,ap=atm)
        
        return  ((ap-vp)*coef(TC))/(adj(TC)+TC)
          
    end
    
    return DO(TC,vp(TC),atm)*(1/1000)*(1/32)*(18/1000)*HC(TC)*atm
end

File: inst/Rust/pO2.rs
use std::env;

fn hc(x: f64) -> f64 {
let hcc =(-0.0000058333*x.powf(3.)+0.0001821*x.powf(2.)+0.072405*x+2.5443)*10000.0;
//println!("hc: {}",hcc);
return hcc;
}

fn vp(x: f64) -> f64 {
let vpp = 0.0456*x.powf(2.)-(0.8559*x)+16.509;
//println!("vp: {}",vpp);
return vpp
}


fn dd(tc: f64,vp: f64,ap: f64)-> f64{
    let coef;
    let adj;
    if tc <30.0 {
    coef=0.678;
    adj = 35.0;
    }else{
    coef=0.827;
    adj = 49.0;
    }
   // println!("coef: {}",coef);
    //println!("adj: {}",adj);
    let ddd= ((ap-vp)*coef)/(adj+tc);
  //  println!("DO: {}",ddd);
    return ddd;
  }


fn main(){
let args: Vec<String> = env::args().collect();
let ref tc0=args[1];
let tc: f64 = tc0.parse().unwrap();
let vvp = vp(tc);
let atm = 760.0;
let ddo = dd(tc,vvp,atm);
let hhc=hc(tc);
println!("pO2: {}",ddo*(1./1000.)*(1./32.)*(18./1000.)*hhc*atm);
}
